### DISCUSSION TITLE
Error "Target 3 is out of bounds"

### DISCUSSION THREAD

**Discussion by Sfonzie98** [pyg-team/pytorch_geometric#4703] [Q&A]
Hello, I got a problem:

I am trying to do a multiclass classification of drug activity using a GCN, I have three activity classes and this is my network:

```
num_classes = 3

class GCN(torch.nn.Module):
  def __init__(self, hidden_channels, hidden_channels1):
    super(GCN, self).__init__()
    self.conv1 = GCNConv(9, hidden_channels, cached=False)
    self.bn1 = BatchNorm1d(hidden_channels)
    self.conv2 = GCNConv(hidden_channels, hidden_channels, cached=False)
    self.bn2 = BatchNorm1d(hidden_channels)
    self.conv3 = GCNConv(hidden_channels, hidden_channels1, cached=False)
    self.bn3 = BatchNorm1d(hidden_channels1)
    self.fc1 = Linear(hidden_channels1, hidden_channels1)
    self.bn3 = BatchNorm1d(hidden_channels1)
    self.fc2 = Linear(hidden_channels1, hidden_channels1)
    self.fc3 = Linear(hidden_channels1, num_classes)
    
  def forward(self, x, edge_index, batch):
    x = F.relu(self.conv1(x, edge_index))
    x = self.bn1(x)
    x = F.relu(self.conv2(x, edge_index))
    x = self.bn2(x)
    x = F.relu(self.conv3(x, edge_index))
    x = self.bn3(x)
    x = global_add_pool(x, batch)
    x = F.relu(self.fc1(x))
    x = self.bn3(x)   
    x = F.relu(self.fc2(x))
    x = F.dropout(x, p=0.26, training=self.training)    
    x = self.fc3(x)
    x = F.log_softmax(x, dim=1)    
    return x
```

When I train the net with `nll_loss` I get this error "Target 3 is out of bounds", I tried to use another loss function (like `MSEloss`) but then I get the error "The size of tensor a (3) must match the size of tensor b (64) at non-singleton dimension 1 "(64 is my batch size). What can I do?

**Comment by rusty1s** [comment]
It looks like your targets do not start from zero. If you have three classes, then `data.y` should only contain values `[0, 1, 2]`.
